llvm/lib/IR/Instruction.cpp

```diff
@@ -177,6 +177,15 @@ void Instruction::setNonNeg(bool b) {
                          (b * PossiblyNonNegInst::NonNeg);
 }
 
+bool Instruction::hasAllowReassocOfAllOperand() const {
+  return all_of(operands(), [](Value *V) {
+    if (!isa<IntrinsicInst>(V))
```

Comment:
reassoc isn't a property of IntrinsicInst, but FPMathOperator

Reply:
Done, updated with `<FPMathOperator>`.